library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableMapImpl.java

```diff
@@ -236,13 +237,27 @@ public void beginWindow(long windowId)
   @Override
   public void endWindow()
   {
+    boolean isTimeUnifiedStore = (store instanceof ManagedTimeUnifiedStateImpl);
     for (K key: cache.getChangedKeys()) {
-      store.put(getBucket(key), keyValueSerdeManager.serializeDataKey(key, true),
+      //the getBucket() returned in fact is time, the bucket assign then assigned the bucketId
+      long timeOrBucketId = getBucket(key);
+      long bucketId = timeOrBucketId;
+      if (isTimeUnifiedStore) {
+        bucketId = ((ManagedTimeUnifiedStateImpl)store).getTimeBucketId(timeOrBucketId);
+      }
+      keyValueSerdeManager.updateBuffersForBucketChange(bucketId);
+      store.put(timeOrBucketId, keyValueSerdeManager.serializeDataKey(key, true),
           keyValueSerdeManager.serializeValue(cache.get(key)));
     }
 
     for (K key: cache.getRemovedKeys()) {
-      store.put(getBucket(key), keyValueSerdeManager.serializeDataKey(key, true), BufferSlice.EMPTY_SLICE);
+      long timeOrBucketId = getBucket(key);
+      long bucketId = timeOrBucketId;
+      if (isTimeUnifiedStore) {
+        bucketId = ((ManagedTimeUnifiedStateImpl)store).getTimeBucketId(timeOrBucketId);
+      }
+      keyValueSerdeManager.updateBuffersForBucketChange(bucketId);
+      store.put(timeOrBucketId, keyValueSerdeManager.serializeDataKey(key, true), BufferSlice.EMPTY_SLICE);
```

Comment:
I not really get the intension or benefit or this section of code. Could you explain more detail